### PR TITLE
Rad station

### PR DESCRIPTION
**Pull request by BriggsIDP**
## About The Pull Request

Rad Station! A new map for Bee station

## Why It's Good For The Game

I've spend a long time working out the layout of the station, planning interactions from the perspective of all payers and jobs, I think this map is both simple to navigate and enjoyable to play!

## Testing Photographs and Procedure

<details>

<summary>Screenshots&Videos</summary>

image

image

image

image

image

image

ion-thruster

</details>

## Changelog
:cl:
add: New map, Radstation
tweak: basemap.dm - added Rad
tweak: maps.txt - added Rad
tweak: decals.dmi - added updated rad sign
imageadd: decals.dmi - added rad shelter sign
tweak: signs_warning.dm - updated rad sign, added rad shelter sign
imageadd: decals.dmi - added minskydorms
imageadd: decals.dmi - added minskyhydrob
imageadd: decals.dmi - added minskyhead
imageadd: decals.dmi - added minskyevac
tweak: signs_departments.dm - added signs listed above
imageadd: 4x7dmi
imageadd: huge ion engine sprite
add: sci smart fridge - for storing components and disks
/:cl:

### CODE REVIEW

**Comment by itsmeow** [comment]
Nooooo not more stations

**Comment by zeskorion** [comment]
the maint doors into viro are weird and clash with random maint door layout conventions 

~~anyways,  you want me to do disposals for you?~~

**Comment by BriggsIDP** [comment]
> the maint doors into viro are weird and clash with random maint door layout conventions
Thanks, I'll look into this again
> ~anyways, you want me to do disposals for you?~
image

**Review by BriggsIDP — COMMENTED** [review]
Thanks for the help Wilson!

**Comment by Fronsis** [comment]
What's the aim of pop for this map?

**Comment by BriggsIDP** [comment]
> What's the aim of pop for this map?

Fairly low. as its not huge. i did aim to give two of everything for jobs. 20 and up i would say

**Comment by Fronsis** [comment]
> image unsure if you should be giving the CE advanced tools

iirc they already start with a toolbelt of advanced tools

**Comment by zeskorion** [comment]
> image

bonuses: no wallpipes
Downsides: you will not be able to read them

**Comment by BriggsIDP** [comment]
Thanks for the feedback spockye!

**Comment by WrillWasTaken** [comment]
the frog in botany is incredibly annoying and pushes you while you're doing things

**Comment by BriggsIDP** [comment]
Of all the departments to screw up I can't believe I did it to botany haha

**Comment by itsmeow** [comment]
Vault needs to be moved to a less secure area. It needs to be somewhat reasonably breachable by your average antag with a welder, currently it's right next to bridge and the captain's office, with no other ways in. Getting to it as an antag would be near impossible without being seen

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm [hunk 1]
```diff
@@ -549,3 +549,20 @@
...
+	desc = "A smart storage vender for tech."
+
+/obj/machinery/smartfridge/sci/accept_check(obj/item/O)
+	if(istype(O, /obj/item/stock_parts/))
+		return TRUE
+	if(istype(O, /obj/item/disk/tech_disk/))
+		return TRUE
+	if(istype(O, /obj/item/circuit_component/))
+		return TRUE
+	if(istype(O, /obj/item/assembly/))
```
Comment (itsmeow) [inline code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm]: trailing `/` on typepaths, remove them

Also this should probably use a typecache, look for other examples of typecaches and use that

File: code/game/shuttle_engines.dm [hunk 2]
```diff
@@ -155,6 +155,26 @@
 	bound_height = 96
 	appearance_flags = 0
 
+/obj/structure/shuttle/engine/hugeionengine
+	name = "engine"
```
Comment (itsmeow) [inline code/game/shuttle_engines.dm]: the name should be clearer and more flavorful, like "bluespace-ion engine"

File: code/game/shuttle_engines.dm [hunk 3]
```diff
@@ -155,6 +155,26 @@
...
+	opacity = 1
```
Comment (itsmeow) [inline code/game/shuttle_engines.dm]: this var is useless, the default opacity is 1

File: code/game/shuttle_engines.dm [hunk 4]
```diff
@@ -155,6 +155,26 @@
...
+	icon = 'icons/obj/4x7.dmi'
+	icon_state = "huge_ion_engine"
+	desc = "An extremely large bluespace engine used to propel even stations."
+	bound_width = 128
+	bound_height = 224
+	appearance_flags = 0
```
Comment (itsmeow) [inline code/game/shuttle_engines.dm]: 0 should be `NONE`. since this is a flags var

File: code/game/shuttle_engines.dm [hunk 5]
```diff
@@ -155,6 +155,26 @@
...
+
+/obj/structure/shuttle/engine/hugeionafterburn
+	name = "Ion"
+	opacity = 1
+	icon = 'icons/obj/4x7.dmi'
+	icon_state = "huge_ion_afterburn"
+	desc = "Quite hot!."
```
Comment (itsmeow) [inline code/game/shuttle_engines.dm]: What even is this one? name and desc need more detail, same reviews as above apply

File: code/game/objects/structures/signs/signs_warning.dm [hunk 6]
```diff
@@ -47,12 +47,6 @@
 	icon_state = "fire"
 	resistance_flags = FIRE_PROOF
 
-/obj/structure/sign/warning/cold
-	name = "\improper DANGER: Extreme cold"
-	desc = "A warning sign which reads 'DANGER: Extreme cold'."
-	icon_state = "cold"
-	resistance_flags = FIRE_PROOF
```
Comment (itsmeow) [inline code/game/objects/structures/signs/signs_warning.dm]: don't remove this

File: code/datums/shuttles.dm [hunk 7]
```diff
@@ -646,6 +646,10 @@
 	suffix = "large"
 	name = "mining shuttle (Large)"
 
+/datum/map_template/shuttle/mining/rad
+	suffix = "rad"
+	name = "mining shuttle (rad)"
```
Comment (itsmeow) [inline code/datums/shuttles.dm]: capitalize (rad)

File: _maps/shuttles/mining/mining_rad [hunk 8]
```diff
@@ -0,0 +1,177 @@
+//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE
```
Comment (itsmeow) [inline _maps/shuttles/mining/mining_rad]: this is not a dmm file, add the dmm extension to the filename

File: _maps/shuttles/mining/mining_rad [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+/obj/structure/ore_box,
+/turf/open/floor/plasteel/tech/grid,
+/area/shuttle/mining)
+"p" = (
+/obj/machinery/door/airlock/survival_pod/glass,
+/obj/docking_port/mobile{
+	dir = 8;
+	dwidth = 3;
+	height = 5;
+	id = "mining";
```
Comment (itsmeow) [inline _maps/shuttles/mining/mining_rad]: these vars need to be absolutely perfect.

dwidth is the horizontal offset OF the docking port from the bottom left corner from the perspective of the docking port's facing dir, height is the vertical offset that the ship ends from the perspective of the docking port, width is the same but horizontal. port_direction and dir determine how it lands

make sure this works

**Review by itsmeow — CHANGES_REQUESTED** [review]
remove box/fland changes

File: _maps/shuttles/mining/mining_rad (hunk 9 above)
Comment (itsmeow) [reply _maps/shuttles/mining/mining_rad]: See: https://wiki.beestation13.com/view/Guide_to_mapping#Dwidth_and_Dheight_in_more_depth

**Comment by itsmeow** [comment]
image
Why is there a random tube that goes directly to chapel?

File: code/game/shuttle_engines.dm (hunk 2 above)
Comment (BriggsIDP) [reply code/game/shuttle_engines.dm]: how is? "Nanotrasen MkIII BPDT engine"

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm (hunk 1 above)
Comment (BriggsIDP) [reply code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm]: like this?
```
/obj/machinery/smartfridge/sci/accept_check(obj/item/O)
	if(typecacheof(list(0, /obj/item/stock_parts, /obj/item/disk/tech_disk, /obj/item/circuit_component, /obj/item/assembly)))
		return TRUE
	return FALSE
```

File: code/game/shuttle_engines.dm (hunk 5 above)
Comment (BriggsIDP) [reply code/game/shuttle_engines.dm]: it's the 'flame' from the engine. I was told to have it as a separate object

**Comment by BriggsIDP** [comment]
> Vault needs to be moved to a less secure area. It needs to be somewhat reasonably breachable by your average antag with a welder, currently it's right next to bridge and the captain's office, with no other ways in. Getting to it as an antag would be near impossible without being seen

new vault location:
image

**Comment by BriggsIDP** [comment]
> image Why is there a random tube that goes directly to chapel? image

removed!

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm [hunk 10]
```diff
@@ -549,3 +549,14 @@
...
 		return FALSE
+
+// ----------------------------
+//  Sci smartfridge
+// ----------------------------
+/obj/machinery/smartfridge/sci
+	desc = "A smart storage vender for tech."
+
+/obj/machinery/smartfridge/sci/accept_check(obj/item/O)
+	if(typecacheof(list(0, /obj/item/stock_parts, /obj/item/disk/tech_disk, /obj/item/circuit_component, /obj/item/assembly)))
```
Comment (itsmeow) [inline code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm]: nono no that's not how you do that

you need to make the typecache statically in the `Initialize()` of the smartfridge, then call `is_in_typecache` on the typecache object

Although as a mapper I don't expect you to know how to do this.. so probably just revert it to the old version but remove the trailing slashes.

File: code/game/shuttle_engines.dm (hunk 4 above)
Comment (itsmeow) [reply code/game/shuttle_engines.dm]: You didn't replace 0 with `NONE`

File: code/game/objects/structures/signs/signs_warning.dm [hunk 11]
```diff
@@ -67,6 +67,11 @@
 	desc = "A warning sign alerting the user of potential radiation hazards."
 	icon_state = "radiation"
 
+/obj/structure/sign/warning/radiation_shelter
+	name = "\improper RADIATION SHELTER ZONE"
+	desc = "A sign alerting the user that this area is shelterd from radiation."
```
Comment (itsmeow) [inline code/game/objects/structures/signs/signs_warning.dm]: "sheltered"

File: code/game/objects/structures/signs/signs_departments.dm [hunk 12]
```diff
@@ -157,6 +162,16 @@
...
 	icon_state = "minskysec"
 
+/obj/structure/sign/departments/minsky/security/command
+	name = "Command Department"
+	desc = "A sign labelling the Command department of the station. Ask the helpful HOP for AA"
+	icon_state = "minskyhead"
+
+/obj/structure/sign/departments/minsky/security/evac
+	name = "Evacuation lounge"
+	desc = "A sign labelling the evacuation lounge of the station. A watch ETA never arrives"
```
Comment (itsmeow) [inline code/game/objects/structures/signs/signs_departments.dm]: these two descs need a period at the end. Also change "watch" to "watched", capitalize "Lounge" in the name

File: code/datums/shuttles.dm (hunk 7 above)
Comment (itsmeow) [reply code/datums/shuttles.dm]: You did not capitalize rad

File: _maps/holodeck/meeting.dmm [hunk 13]
```diff
@@ -0,0 +1,456 @@
+//MAP CONVERTED BY dmm2tgm.py THIS HEADER COMMENT PREVENTS RECONVERSION, DO NOT REMOVE
+"a" = (
+/obj/effect/turf_decal/siding/wood/corner{
+	dir = 1
+	},
+/turf/open/floor/carpet/royalblue,
+/area/template_noop)
+"b" = (
```
Comment (itsmeow) [inline _maps/holodeck/meeting.dmm:8]: why is this here? you need to add the template for this to show up in game if that was what you wanted

**Review by itsmeow — CHANGES_REQUESTED** [review]

File: code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm (hunk 10 above)
Comment (BriggsIDP) [reply code/modules/food_and_drinks/kitchen_machinery/smartfridge.dm]: this hurts my brain, but will do!

**Review by itsmeow — COMMENTED** [review]
Code approved

**Comment by Programs-The-Station** [comment]
Issues found in Round 43196:

image
image
Both incinerator shutters are controlled via one button.
image
Telecomm lobby has no vent, so it's vacuum round start.

**Review by spockye — COMMENTED** [review]
---
ignore this one
---

image
missing vent

image

missing pipe

image

missing pipe (ai core)

image

cable wackyness

image

missing cables

image

missing cable

image
missing cable

image

wacky cable

image

disposals wackery (med)

image

disposals wackery (chapel)

image

outside atmos

image

disposals wackery (xeno)

image

it stands, in complete solitude. Devoid of any contact to it's kin. (sci)

image

why is there escape pod zones on the escape pod docks?

image
passthrough zones

image

unzoned, floating walls

image

give solars solars zoning

image

zoning on space

image
theres a small problem with doing this. A lot of random rooms have wallmounts, and they'll end up floating. it might be best to just do static maints here?

image

Rotate it

image

these are walled in

image

get it out of the wall

why does the captain and hop share an office now?

image
rotate these

image

don't pixelshift chairs

image

don't pixelshift chairs

image

don't pixelshift chairs

image

you should generally try to keep wallmounts off windows (you can't put them back on)
the fire alarm is rotated the wrong way


im realizing that i might be on a old version of the map, do scream at me if that's the case

**Review by spockye — COMMENTED** [review]
image

missing pipe

image

pipe quirkyness (bridge)

image

pipe quirkyness (sec)

image

pipe quirkyness (outside SM)

image

no idea what this is

image

toxins

image

library

image

Departures

image

engineering

image

disposals quirkyness (bridge)

image

Hos office

image

Chapel

image

Xeno

image

Chem, Med maintenance, Med storage

---

image

probably best to just replace this with a normal wall

image

theres no need for shuttle zoning on the docks. Look at what other maps do

image

Floating walls with no zoning

image

Rotate these!

image

These are blocked in. See if you can find a way to open it up!

( i forgot to update my version. OOPS)

**Comment by itsmeow** [comment]
image
Goofy ahh maint room door/APC placement, table in tha way (blurred for privacy)

Probably best to make this a static maint room

**Review by spockye — COMMENTED** [review]
image

zone weirdness (starboard quarter solar)

image

remove this zoning weirdness

image

missing zone

image

i'd recommend moving the layers of the railings to 3.1, so that the machinery/people/chairs go under it!

File: code/game/shuttle_engines.dm [hunk 14]
```diff
@@ -156,6 +156,25 @@
...
+	name = "Nanotrasen MkIII BPDT engine"
+	icon = 'icons/obj/4x7.dmi'
+	icon_state = "huge_ion_engine"
+	desc = "An extremely large bluespaceoplasmadynamic ion engine used to propel even stations."
+	bound_width = 128
+	bound_height = 224
+	appearance_flags = NONE
+
+/obj/structure/shuttle/engine/hugeionafterburn
+	name = "Nanotrasen MkIII BPDT engine afterburn"
```
Comment (itsmeow) [inline code/game/shuttle_engines.dm]:
```suggestion
	name = "Nanotrasen MkIII BPDT engine afterburner"
```

File: code/game/shuttle_engines.dm [hunk 15]
```diff
@@ -156,6 +156,25 @@
 	bound_height = 96
 	appearance_flags = 0
 
+/obj/structure/shuttle/engine/hugeionengine
+	name = "Nanotrasen MkIII BPDT engine"
+	icon = 'icons/obj/4x7.dmi'
+	icon_state = "huge_ion_engine"
+	desc = "An extremely large bluespaceoplasmadynamic ion engine used to propel even stations."
```
Comment (itsmeow) [inline code/game/shuttle_engines.dm]:
```suggestion
	desc = "An extremely large bluespace-plasmadynamic ion engine used to propel objects reaching the size of stations."
```

File: code/game/shuttle_engines.dm [hunk 16]
```diff
@@ -156,6 +156,25 @@
...
+	bound_width = 128
+	bound_height = 224
+	appearance_flags = NONE
+
+/obj/structure/shuttle/engine/hugeionafterburn
+	name = "Nanotrasen MkIII BPDT engine afterburn"
+	opacity = 1
+	icon = 'icons/obj/4x7.dmi'
+	icon_state = "huge_ion_afterburn"
+	desc = "Quite hot, don't get too close to the glowing end!."
```
Comment (itsmeow) [inline code/game/shuttle_engines.dm]:
```suggestion
	desc = "Quite hot, don't get too close to the glowing end!"
```

**Review by itsmeow — CHANGES_REQUESTED** [review]
add Rad to the README list of available maps

**Review by PowerfulBacon — APPROVED** [review]

**Review by itsmeow — APPROVED** [review]
